environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -152,6 +154,11 @@ static WeldContainer initialize(String id, WeldManager manager, Bootstrap bootst
         return weldContainer;
     }
 
+    private static boolean isShutdownHookNeeded() {
+        String property = AccessController.doPrivileged(new GetSystemPropertyAction(Weld.SHUTDOWN_HOOK_SYSTEM_PROPERTY));
+        return property == null || Boolean.FALSE.equals(Boolean.valueOf(property));
```

Comment:
No, `org.jboss.weld.se.shutdownHook=true` means *skip* the registration. I will change the system property name... this one is confusing. Thanks for feedback!